src/components/menu/menu.component.ts

```diff
@@ -29,11 +29,14 @@ export default class SlMenu extends ShoelaceElement {
   }
 
   private handleClick(event: MouseEvent) {
-    if (!(event.target instanceof SlMenuItem)) {
-      return;
-    }
+    const menuItemTypes = ["menuitem", "menuitemcheckbox"]
+
+    const target = event.composedPath().find((el: Element) => menuItemTypes.includes(el?.getAttribute?.("role") || ""))
```

Comment:
Wouldn't using the tag name for `sl-menu-item` be less ambiguous?

Reply:
@diervo I want to belive in a future where we can have arbitrary elements slotted in that conform to a certain API 🙈 I know its kind of lying to the TS compiler...but perhaps that is the correct solution here to rely on `sl-menu-item` tag.